src/python/pants/backend/docker/target_types.py

```diff
@@ -305,11 +305,17 @@ def option_values(self, **kwargs) -> Iterator[str]:
         # `path` is an absolute path, the `buildroot` and `spec_path` will not be considered.  Also,
         # an empty path part is ignored.
         for secret, path in (self.value or {}).items():
-            full_path = os.path.join(
-                get_buildroot(),
-                self.address.spec_path if re.match(r"\.{1,2}/", path) else "",
-                path,
-            )
+            if os.path.isabs(path):
+                full_path = path
+            elif path != os.path.expanduser(path):
+                full_path = os.path.expanduser(path)
+            else:
+                full_path = os.path.join(
+                    get_buildroot(),
+                    self.address.spec_path if re.match(r"\.{1,2}/", path) else "",
+                    path,
+                )
```

Comment:
I think a smaller change would be to only add the expanduser call to path directly and thus avoid the if-else contructs:
```suggestion
                full_path = os.path.join(
                    get_buildroot(),
                    self.address.spec_path if re.match(r"\.{1,2}/", path) else "",
                    os.path.expanduser(path),
                )
```

Reply:
Thanks for the quick review. Updated the help and simplified the code [here](https://github.com/pantsbuild/pants/pull/19130/commits/74729513614560531bd9585f1a45146ac3c0c122)